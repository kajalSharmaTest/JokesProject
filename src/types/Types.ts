export interface Joke {
  id: string;
  joke: string;
}
export interface IntialState {
  jokes?: Array<Joke>;
  page: number;
  loading: boolean;
  error?: string;
}
export interface Action {
  type: string;
  payload?: Array<Joke>;
}
