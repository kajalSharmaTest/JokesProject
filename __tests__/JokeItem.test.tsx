import 'react-native';
import React from 'react';
import JokesItem from '../src/Components/JokeItem';

import renderer from 'react-test-renderer';
import {Joke} from '../src/types/Types';

const joke: Joke = {
  id: '1',
  joke: 'Test joke',
};

describe('Joke item test', () => {
  it('should render correctly', () => {
    // passing mock joke object as prop in JokeElement component.
    const {toJSON} = renderer.create(<JokesItem joke={joke} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
