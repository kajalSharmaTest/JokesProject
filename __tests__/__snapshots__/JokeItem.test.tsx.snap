// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke item test should render correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#FFA500",
      "borderRadius": 10,
      "borderWidth": 1,
      "flex": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    Test joke
  </Text>
</View>
`;
